index.html

```diff
@@ -0,0 +1,475 @@
+
+### Portability
+A type is *portable to* a subgraph if it has a `@csdl_key` for that subgraph.
+
+### Free / Bound fields
+
+Free fields can be resolved by any subgraph, bound fields can only be resolved by the subgraph they're bound to.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
+
+```graphql
+#::[Using `csdl_Graph` to define endpoints]
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+```
+
+The `csdl_Graph` enum is used as input to csdl directives which link fields and types to subgraphs.
+
+## Provided Types
+
+Valid CSDL documents **must** include the declarations of any CSDL types they use.
+
+### csdl_Never
+```graphql
+#::[Declaration of csdl_Never]
+type csdl_Never {}
+```
+
+The `csdl_Never` type is used to indicate the type of a field which **cannot be resolved**. This is typically due to a composition error. Producers **should** provide more specific error information with a `@csdl_Error` directive on the relevant field.
+
+```graphql
+#::[Using `csdl_Never` and `@csdl_error` to mark a field error]
+type SomeType {
+  someField: csdl_Never
+    @csdl_error(message: "someField defined with conflicting types: Int, String")
+}
+```
+
+## Scalars
+
+Valid CSDL documents **must** include the declarations of any CSDL scalars they use. Producers **should** include the `@specifiedBy` directive, but it is not mandatory.
+
+### `csdl_SelectionSet`
+
+```graphql
+#::[`scalar csdl_SelectionSet` declaration]
+scalar csdl_SelectionSet
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_selectionset")
+```
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+All legal selection set features are allowed. That includes alises, field arguments, directives, and inline fragments. Named fragment spreads are syntactically valid, but meaningless, as there is nowhere to define named fragments; future versions of this spec may change this.
+
+### `csdl_URL`
+
+```graphql
+#::[`scalar csdl_URL` declaration]
+scalar csdl_URL
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_url")
+```
+
+A [URL](https://www.w3.org/Addressing/URL/url-spec.html). Used by `@csdl_endpoint`.
+
+## Directives
+
+Valid CSDL documents **must** include the declarations of any CSDL directives they use.
+
+### `@csdl_key`
+
+```graphql
+#::[`@csdl_key` declaration]
+directive @csdl_key(graph: csdl_Graph!, repr: csdl_SelectionSet!)
```

Comment:
Perhaps the first argument should be `subgraph`?

Not sure I like `repr` as the second argument. repr, I think, means the argument to `Query._entities`. But this string is not a repr: it's a selection set that can be evaluated to produce a repr.

Reply:
I still like `fields` for this. It is true that selections may contain inline fragments in addition to fields, but those get reduced to fields once we know the runtime type we're computing an entity for. It also makes it clear this directly corresponds to `fields` in the user-facing `@key` directive.